File: app/src/main/java/com/stan/dagger_android/home/ActivityModule.java
package com.stan.dagger_android.home;


import com.stan.dagger_android.data.Student;
import com.stan.dagger_android.di.ActivityScoped;
import com.stan.dagger_android.di.FragmentScoped;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;
import dagger.android.ContributesAndroidInjector;

/**
 * Created by uu on 2017/12/28.
 */
@Module
public abstract class ActivityModule {

    @Provides
    @ActivityScoped
    static Student getStudent() {
        return new Student("XiaoHong");
    }

    @FragmentScoped
    @ContributesAndroidInjector
    abstract MainFragment addMainFragment();

    @ActivityScoped
    @Binds
    abstract MainActivityContract.Presenter mainPresenter(MainActivityPresenter presenter);
}

File: app/src/main/java/com/stan/dagger_android/home/ActivityModule.kt
package com.stan.dagger_android.home


import com.stan.dagger_android.data.Student
import com.stan.dagger_android.di.ActivityScoped
import com.stan.dagger_android.di.FragmentScoped

import dagger.Binds
import dagger.Module
import dagger.Provides
import dagger.android.ContributesAndroidInjector

/**
 * Created by uu on 2017/12/28.
 */
@Module
abstract class ActivityModule {

    @FragmentScoped
    @ContributesAndroidInjector
    internal abstract fun addMainFragment(): MainFragment

    @ActivityScoped
    @Binds
    internal abstract fun mainPresenter(presenter: MainActivityPresenter): MainActivityContract.Presenter

    @Module
    companion object {
        @Provides
        @JvmStatic
        @ActivityScoped
        internal fun student(): Student{
            return  Student("XiaoHong")
        }
    }
}
